src/rest_vol.c

```diff
@@ -293,25 +294,25 @@ H5rest_init(void)
     if (H5open() < 0)
         FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTINIT, FAIL, "HDF5 failed to initialize");
 
-    if (H5_rest_id_g >= 0 && (idType = H5Iget_type(H5_rest_id_g)) < 0)
```

Comment:
This line should probably stay here. The VOL connector could possibly be initialized by env. variable rather than H5rest_init, so this line attempts to check if the connector is already registered before doing it again. We could keep a second copy inside the `H5I_VOL != idType` block if we want to be extra sure, but this first instance of it is useful.